### PR TITLE
DJ May 2021

### PR DESCRIPTION
**Pull request by xaur**
**DRAFT**: https://github.com/xaur/decred-news/blob/draft202105/journal/202105.md

**Dependencies**: Send notice in advance so that people can plan their work.

- [ ] [title image](https://github.com/decred/dcrdesign/issues) (notify by: 20th, ready by: 3rd)
- [ ] first time contributors
- [ ] PR update (notify by: 28th, ready by: 3rd)

**Monthly snapshots**: Snapshot data that is hard to obtain later. Aim to capture it at 1st day of the month at 00:00 UTC.

- [ ] social media stats
  - accs reported in People ("key" list)
  - all accs in [social-media-stats](https://github.com/decredcommunity/social-media-stats)
  - untracked candidate accounts ("maybe" list)
  - see if any new accounts need to be tracked
- [ ] Medium and Publish0x post stats
- [ ] Reddit traffic stats
- [ ] mining
  - pool hashrate absolute Ph/s and share %  values
  - [miningpoolstats.stream](https://miningpoolstats.stream/decred)
  - [dcrstats.com/pow](https://dcrstats.com/pow)
  - luxor.tech API
  - save raw data where possible (JSON etc)
- [ ] staking
  - ticket price [30-day average](https://dcrstats.com)
- [ ] network stats
  - nodes at [Decred Mapper](https://nodes.jholdstock.uk/user_agents)
  - nodes at [dcrextdata](https://dcrextdata.planetdecred.org/nodes)
  - update the list of known [nodes and groups](https://github.com/decredcommunity/network-stats/tree/master/nodes)
- [ ] vspd and dcrstakepool instance stats
- [ ] [LN stats](https://ln-map.jholdstock.uk/)

**Writing**

- [ ] extract more dev updates from GitHub repos
- [ ] complete Development (by: 2nd)
- [ ] complete People (by: 2nd)
- [ ] PO review Development and People
- [ ] ask in #dev to review Development and People
- [ ] complete Events
- [ ] (opt) ask #events to review Events
- [ ] complete Outreach
- [ ] complete Media
- [ ] (opt) ask #marketing to review Outreach+Events+Media
- [ ] complete Governance
- [ ] complete Network
- [ ] complete Integrations, Adoption
- [ ] (opt) find interesting Reddit and Twitter threads
- [ ] complete Community Discussions
- [ ] complete Markets
- [ ] complete Relevant External
- [ ] (opt) ask quoted people to review their texts
- [ ] complete Intro
- [ ] PO review of all sections
- [ ] (opt) ask #writers to review (by: 3rd)
- [ ] incorporate all feedback
- [ ] resolve all [TODOs](https://github.com/xaur/decred-news/blob/docs/guidelines.md#todos)
- [ ] ensure that all people who contributed or gave feedback are in [credits](https://github.com/xaur/decred-news/blob/docs/guidelines.md#how-to-give-credit)
- [ ] [lint](https://github.com/xaur/decred-news/blob/docs/guidelines.md#linting)
- [ ] add [title image](https://github.com/xaur/decred-news/blob/docs/guidelines.md#title-image)
- [ ] remove `{DRAFT}` from title
- [ ] pre-release review: check spelling, grammar, test that links work

**Publishing**

- [ ] publish on GitHub with a single commit to `gh-pages`
- [ ] add all images to the [files repo](https://github.com/xaur/decred-journal-files)
- [ ] publish on Medium
  - upload images
  - make Table of Contents
  - add to /decred/journals
  - add relevant tags
  - verify text previews shown at /decred and /decred/journals
  - check for formatting issues
- [ ] publish on Publish0x
  - upload images
  - add relevant tags
  - check for formatting issues
- [ ] post on r/decred
- [ ] tweet via decredproject
  - add title image
- [ ] submit new issue to decred.org Press page ([example](https://github.com/decred/dcrweb/pull/898))
- [ ] if changes are necessary after the release, add commits to `gh-pages` (test in `draftXX` if needed) and update other places
- [ ] announce on: Matrix #dcr, t.me/DCRann, Blockfolio, CoinGecko, LinkedIn
- [ ] [update index](https://github.com/xaur/decred-news/blob/docs/guidelines.md#updating-index)
- [ ] fast-forward `master` to `gh-pages`

**Housekeeping**

- [ ] archive draft history
  - create `tmp` at `master`, merge both `drafts` and the latest draft commit, fast-forward `drafts` to `tmp`
- [ ] (opt) update `journal-template.md` in `docs` branch
- [ ] (opt) update this `release-checklist-template.md` in the `docs` branch
- [ ] start new cycle
- [ ] delete the draft branch (will auto-close the PR)

### CODE REVIEW

File: journal/202105.md [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+
+- added block explorer [links](https://github.com/decred/vspd/pull/247) for ticket and fee transactions to the UI
+- added best block [height](https://github.com/decred/vspd/pull/254) to status response (useful for detecting stalled VSPs)
+- populate ticket [purchase height](https://github.com/decred/vspd/pull/250) for all confirmed tickets and show it on the admin page
+- improved admin page [aesthetics](https://github.com/decred/vspd/pull/263)
+- added a framework for database [upgrades](https://github.com/decred/vspd/pull/242)
+- improved logging when [recovering](https://github.com/decred/vspd/pull/255) from request handling failures
+- added [config](https://github.com/decred/vspd/pull/249) params for max log size and max log files to keep
+- [remove](https://github.com/decred/vspd/pull/260) confirmed fee transactions from the database, cutting disk usage [by ~2x](https://github.com/decred/vspd/pull/243#issuecomment-844945549)
+- store each ticket in its own [DB bucket](https://github.com/decred/vspd/pull/243). Combined with the previous optimization, it allows to trade ~40% higer disk usage for ~50% faster insertion and ~85% faster counting of tickets.
```
Comment (jholdstock) [inline journal/202105.md:143]: Not just counting tickets, but any operation which requires iterating over all tickets - eg. searching/filtering. Maybe go with "~85% faster iterating over tickets."?

File: journal/202105.md [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+
+## Development
+
+The work reported below has the "merged to master" status unless noted otherwise. It means that the work is completed, reviewed, and integrated into the source code that advanced users can [build and run](https://medium.com/@artikozel/the-decred-node-back-to-the-source-part-one-27d4576e7e1c), but is not yet available in release binaries for regular users.
+
+<a id="dcrd" />
+
+**[dcrd](https://github.com/decred/dcrd)**
+
+- moved code for handling [signing](https://github.com/decred/dcrd/pull/2642) standard scripts to its own subpackage, in order to prepare for future splitting of standard script handling from consensus critical code
```
Comment (davecgh) [inline journal/202105.md:55]:
```suggestion
- moved code for handling [signing](https://github.com/decred/dcrd/pull/2642) standard scripts to its own subpackage in order to prepare for future splitting of standard script handling from consensus critical code
```

**Review by davecgh — COMMENTED** [review]
dcrd sections looks accurate.

**Comment by lukebp** [comment]
Politeia section lgtm.

File: journal/202105.md [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+- implemented the [Max](https://github.com/planetdecred/godcr/pull/396) button for setting the maximum DCR amount that can be sent
+- added the submission of input fields with pressing [Enter](https://github.com/planetdecred/godcr/pull/414)
+- added [loading](https://github.com/planetdecred/godcr/pull/400) animations to certain modals
+- initial [dark mode](https://github.com/planetdecred/godcr/pull/401) suppot
+- added language translation infrastructure
+- ~9 bug fixes
+
+Elias Naur (creator of the Gio library that powers GoDCR) has kindly reviewed the UI code and shared his [recommendations](https://paste.sr.ht/~eliasnaur/cea1d29d6a5f96668b5e166c2f39ef596974574f), many of which have already been implemented. { can we link this publicly? }
+
+{ what is the "product name" for use in texts, "godcr" or "GoDCR"? }
```
Comment (oshorefueled) [inline journal/202105.md:228]: godcr

File: journal/202105.md [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+
+{ maybe GoDCR shot, 768px wide if possible }
```
Comment (oshorefueled) [inline journal/202105.md:230]: Screenshot 2021-06-02 at 00 07 31

File: journal/202105.md [hunk 5]
```diff
@@ -0,0 +1,475 @@
...
+
+<a id="godcr" />
+
+**[godcr](https://github.com/planetdecred/godcr)**
+
+- implemented pages: [Tickets](https://github.com/planetdecred/godcr/pull/395) overview, [Tickets list](https://github.com/planetdecred/godcr/pull/406), and [Restore wallet](https://github.com/planetdecred/godcr/pull/386)
+- implemented the [Max](https://github.com/planetdecred/godcr/pull/396) button for setting the maximum DCR amount that can be sent
+- added the submission of input fields with pressing [Enter](https://github.com/planetdecred/godcr/pull/414)
+- added [loading](https://github.com/planetdecred/godcr/pull/400) animations to certain modals
+- initial [dark mode](https://github.com/planetdecred/godcr/pull/401) suppot
```
Comment (oshorefueled) [inline journal/202105.md:222]:
```suggestion
- initial [dark mode](https://github.com/planetdecred/godcr/pull/401) support
```

File: journal/202105.md (hunk 2 above)
Comment (xaur) [reply journal/202105.md]: Applied manually, thanks.

File: journal/202105.md (hunk 1 above)
Comment (xaur) [reply journal/202105.md]: That will work, changed. Thanks for the suggestion.

File: journal/202105.md [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+- added language translation infrastructure
+- ~9 bug fixes
+
+Elias Naur (creator of the Gio library that powers GoDCR) has kindly reviewed the UI code and shared his [recommendations](https://paste.sr.ht/~eliasnaur/cea1d29d6a5f96668b5e166c2f39ef596974574f), many of which have already been implemented. { can we link this publicly? }
```
[the remark this answers is not in the document]
Comment (xaur) [reply journal/202105.md]: Added with minor text tweaks, thanks.

File: journal/202105.md (hunk 5 above)
Comment (xaur) [reply journal/202105.md]: Corrected.

File: journal/202105.md (hunk 3 above)
Comment (xaur) [reply journal/202105.md]: Just to clarify, I refer to the "app display name" and not development/repository name. For example, we use `DCRDEX` as a display name while `dcrdex` is the repo name.

I think I got confused by "GoDCR" in the [proposal](https://proposals.decred.org/proposals/e5c8051), which later leaked to Twitter.

 I see that `godcr` is in the title bar, so will use that. Thanks.

File: journal/202105.md (hunk 4 above)
Comment (xaur) [reply journal/202105.md]: Thanks for sharing the image! I'll add it with the other images at a later stage of the release cycle.

**Pull request closed without merging**